Dockerfile

```diff
@@ -114,24 +115,24 @@ RUN apk add --no-cache --virtual=.build-deps build-base && \
      apk del .build-deps && \
      mkdir -p /tmp/.cache
 
+# TODO: we should remove this (we were supposed to get rid of it in June 2022)
 COPY entrypoint.sh /entrypoint.sh
 RUN chmod +x /entrypoint.sh
 
 # Let the user know how their container was built
 COPY Dockerfile /Dockerfile
 
 # Get semgrep-core from step1
-COPY --from=semgrep-core /semgrep/semgrep-core/_build/default/src/cli/Main.exe /usr/local/bin/semgrep-core
+COPY --from=semgrep-core-container /src/semgrep/semgrep-core/_build/default/src/cli/Main.exe /usr/local/bin/semgrep-core
 
+# ???
 ENV SEMGREP_IN_DOCKER=1 \
      SEMGREP_VERSION_CACHE_PATH=/tmp/.cache/semgrep_version \
      SEMGREP_USER_AGENT_APPEND="Docker"
 
-WORKDIR /src
-
 # /semgrep is not needed anymore;
 # semgrep is now available /usr/local/bin thx to the 'pip install' command above
-# (this weirdly does not reduce the size of the docker image though)
+# TODO: this weirdly does not reduce the size of the docker image though
 RUN rm -rf /semgrep
```

Comment:
It's a good observation. This is how docker images work. The state of the filesystem after each docker instruction is called a layer and remains available in the final image, similarly to diffs in a git history. To save space, we'd have to start another docker build stage like we already [do](https://github.com/returntocorp/semgrep/blob/3b82b71d493d5eaa367e3eec6c8531a08d4bd7f2/Dockerfile#L83) between the ocaml build and the python build. See the [Docker documentation on multistage builds](https://docs.docker.com/build/building/multi-stage/).